extensions/kubernetes/runtime/src/main/java/org/apache/camel/quarkus/component/kubernetes/cluster/KubernetesClusterServiceConfig.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.kubernetes.cluster;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BooleanSupplier;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import org.apache.camel.component.kubernetes.cluster.LeaseResourceType;
+import org.eclipse.microprofile.config.ConfigProvider;
+
+@ConfigRoot(name = "camel.cluster.kubernetes")
+public class KubernetesClusterServiceConfig {
+
+    /**
+     * Whether a Kubernetes Cluster Service should be automatic configured
+     * according to 'quarkus.camel.cluster.kubernetes.*' configurations.
+     */
+    @ConfigItem
+    public Optional<Boolean> enabled;
+
+    /**
+     * The name of the lease resource used to do optimistic locking (defaults to
+     * 'leaders'). The resource name is used as prefix when the underlying
+     * Kubernetes resource can manage a single lock.
+     */
+    @ConfigItem
+    public Optional<String> kubernetesResourceName;
+
+    /**
+     * The lease resource type used in Kubernetes, either 'config-map' or
+     * 'lease' (defaults to 'lease').
+     */
+    @ConfigItem
+    public Optional<LeaseResourceType> leaseResourceType;
+
+    /**
+     * Whether the Camel cluster namespaces should be distributed evenly across
+     * the kubernetes nodes. Either 'true' or 'false' (defaults to 'true').
+     */
```

Comment:
@lburgazzoli @jamesnetherton Is the documentation above about rebalancing the right one ?

Reply:
Maybe it can reflect the description here:; https://github.com/apache/camel/blob/main/core/camel-support/src/main/java/org/apache/camel/support/cluster/RebalancingCamelClusterService.java#L38-L43

Reply:
This is what I intend :) I was hoping to have it translated into simpler words. Maybe I should look deeper the whole hierarchy and tests.